sdk/core/Azure.Core/src/Shared/MessagingClientDiagnostics.cs

```diff
@@ -170,10 +170,10 @@ public void InstrumentMessage(IDictionary<string, object> properties, string act
                     DiagnosticScope.ActivityKind.Producer);
                 messageScope.Start();
 
-                Activity activity = Activity.Current;
+                Activity? activity = Activity.Current;
                 if (activity != null)
                 {
-                    traceparent = activity.Id;
+                    traceparent = activity!.Id!;
```

Comment:
Is the `!` after `activity` needed, considering this line is wrapped in a null check?

Reply:
Ah I don't think so. I fixed it.